airflow/www/async_dag_loaders.py

```diff
@@ -0,0 +1,200 @@
+    """A process that creates, updates, and sync dagbag in background."""
+    import airflow
+    # pylint: disable=redefined-outer-name,reimported
+    from airflow import configuration
+
+    # TODO(coufon): env vars that are not exposed to subprocesses are invisible here.
+    # In Composer we reload all env vars here, should check whether a similar step may
+    # be needed.
+
+    configuration = importlib.reload(configuration)
+    airflow.configuration = importlib.reload(airflow.configuration)
+    airflow.plugins_manager = importlib.reload(airflow.plugins_manager)
+    airflow = importlib.reload(airflow)
+
+    signal.signal(signal.SIGTERM, _kill_proc)
+    signal.signal(signal.SIGINT, _kill_proc)
+
+    logging.info('Using Asynchronous Dagbag Loader.')
+    dagbag = DagBag(dag_folder)
+    event_collect_done = Event()
+    event_next_collect = Event()
+    thread = threading.Thread(target=_send_dagbag,
+                              args=(dagbag, queue, event_collect_done,
+                                    event_next_collect))
+    thread.daemon = True
+    thread.start()
+    while True:
+        try:
+            event_collect_done.clear()
+            event_next_collect.clear()
+
+            start_time = time.time()
+            dagbag.collect_dags(dag_folder)
+            event_collect_done.set()
+            event_next_collect.wait()
+
+            time.sleep(max(0, COLLECT_DAGS_INTERVAL - (time.time() - start_time)))
+        except SystemExit:
+            sys.exit(0)
+        except Exception:  # pylint: disable=broad-except
+            logging.warning('Dagbag loader dags collector errors.', exc_info=True)
+
+
+def _cleanup_dagbag(dagbag, current_keys):
+    """Delete keys in the dagbag but not in the latest collection."""
+    for k in ['dags', 'file_last_changed', 'import_errors']:
+        v = dagbag.__dict__[k]
+        for key_to_delete in set(v.keys()) - set(current_keys[k]):
+            del v[key_to_delete]
+        current_keys[k] = []
+
+
+def _receive_dagbag(dagbag, queue):
+    """A thread that receives updated dagbag."""
+    current_keys = defaultdict(list)
+    while True:
+        try:
+            collect_done, dagbag_update = queue.get()
+            for k, v in dagbag_update.items():
+                current_keys[k].extend(v.keys())
+                dagbag.__dict__[k].update(v)
+            if collect_done:
+                _cleanup_dagbag(dagbag, current_keys)
+        except Exception:  # pylint: disable=broad-except
+            logging.warning('Dagbag loader receiver errors.', exc_info=True)
+
+
+def create_async_dagbag(dag_folder):
+    """
+    It creates a new process to collect DAGs with a sender thread puts updated DAGs
+    on a queue. The main process creates a receiver thread to get DAGs and update
+    the DagBag. All new processe and threads are daemon so the main process never
+    waits for them at exiting time.
+    """
+    dagbag = DagBag(dag_folder)
```

Comment:
Doesn't this line load all the dags since `collect_dags` defaults to True

Reply:
Fixed.

Sorry it was a mistake. I use _DagBag inheriting from models.DagBag with overloaded __init__ in the first version, and did not set collect_dags=False in this new version.